app/decorators/service_provider_session_decorator.rb

```diff
@@ -42,8 +54,8 @@ def sp_name
   end
 
   def sp_return_url
-    if request_url.present?
-      OpenidConnectRedirector.from_request_url(request_url).decline_redirect_uri
+    if sp.redirect_uri.present? && openid_connect_redirector.valid?
```

Comment:
@zachmargolis I can't seem to get away from using `sp.redirect_uri.present?`. If`request_url` gets passed to `OpenidConnectRedirector.from_request_url` the URI service throws and error when it tries to parse a non OIDC return url: `bad argument (expected URI object or URI string)`. Would it be best to use a Rescue in this case? Or is there another way to test the `request_url` after its already been sent to `OpenidConnectRedirector`?